sdks/python/apache_beam/pipeline.py

```diff
@@ -891,8 +891,10 @@ def visit_transform(self, transform_node):
             if (isinstance(output.element_type,
                            typehints.TupleHint.TupleConstraint) and
                 len(output.element_type.tuple_types) == 2):
-              output.requires_deterministic_key_coder = (
-                  deterministic_key_coders and transform_node.full_label)
+              if (pcoll.element_type.tuple_types[0] ==
```

Comment:
Should this "if" become an "and" if the existing if above?

Reply:
Yes, that's better, done. (The current structure is an artifact of debugging.)